django/contrib/admin/options.py

```diff
@@ -613,8 +614,14 @@ def get_form(self, request, obj=None, **kwargs):
         # if exclude is an empty list we pass None to be consistent with the
         # default on modelform_factory
         exclude = exclude or None
+
+        # Cancel out overridden form fields which are in readonly_fields.
+        new_attrs = OrderedDict([(f, None) for f in readonly_fields
```

Comment:
I don't think keeping the attributes as `None` on the modelform is the right approach. I think they shouldn't be there at all.

Furthermore, we use hanging indents or keep everything in one line if it doesn't exceed 119 characters (I don't think you need the explicit list in there, a generator should work:
```python
new_attrs = OrderedDict(
    (f, None) for f in readonly_fields
    if f in self.form.declared_fields
)
```